### PR TITLE
Suspicious request blocking tests for Java

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

File: tests/appsec/test_blocking_addresses.py [hunk 1]
```diff
@@ -523,14 +549,17 @@ def setup_non_blocking(self):
...
         )
+        self.rm_req_nonblock3 = weblog.post("/waf", data={"good": "value"})
 
     def test_non_blocking(self):
         """Test if requests that should not be blocked are not blocked"""
         for response in (self.rm_req_nonblock1, self.rm_req_nonblock2):
-            assert response.status_code == 200
+            # Some frameworks will correctly bark 415 at invalid urlencoded-data
+            assert response.status_code in (200, 415)
+        assert self.rm_req_nonblock3.status_code == 200
```
[the remark this answers is not in the document]
Comment (smola) [reply tests/appsec/test_blocking_addresses.py:559]: `application/octet-stream` and `text/plain`, I'm changing the relevant endpoints to allow both, so they'll be 200 again.

[the remark this answers is not in the document]
Comment (smola) [reply tests/appsec/test_blocking_addresses.py]: Based on an offline discussion, I have added endpoints that accept application/octet-stream and text/plain for jax-rs. So the 415 case is gone.

However, in jax-rs, depending on the annotations, we'll treat text/plain as parsed body and block based on it. I have moved this case to a separate case and marked as irrelevant for jax-rs, as it seems we need some further discussion on this behavior.

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI is failing on activated test

**Comment by smola** [comment]
@cbeauchesne It's just that dd-trace-java snapshots stopped being updated in ghcr.io for a couple of weeks, I'm looking for someone who can fix this.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**